### PR TITLE
Toggling AM/PM

### PR DESCRIPTION
**Pull request by jrchudy**
This PR fixes issue #1039. There was a condition for when the page was initialized to set a default value for some fields (if it was available) in the `entry/create` context. If that default wasn't available (it's a function) or it's `null`, `timestamp` and `timestamptz` subfields were initialized to `null` with a default meridiem of `AM`. 

This was then being overwritten to the default value. That should have been nested in an `else` case.

### CODE REVIEW

**Comment by RFSH** [comment]
The change looks good and fixes the issue. I just have a question about [the test case that you changed](https://github.com/informatics-isi-edu/chaise/blob/54350aeea685f093f396d37e57b4d098646c78fa/test/e2e/specs/recordedit/helpers.js#L552). 
Is it possible to break it into different test cases? I know that time limit is a concern for not breaking it into different test cases. But looking at the test case, it's hard to follow a 100+ lines unit test and understand what exactly is trying to test.

**Comment by jrchudy** [comment]
After reviewing the structure of the e2e tests, it isn't possible to factor this section into smaller pieces given the way protractor scopes variables.
